packages/medusa/src/api-v2/store/payment-collections/[id]/payment-sessions/route.ts

```diff
@@ -0,0 +1,64 @@
+import { createPaymentSessionsWorkflow } from "@medusajs/core-flows"
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { IPaymentModuleService } from "@medusajs/types"
+import { remoteQueryObjectFromString } from "@medusajs/utils"
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "../../../../../types/routing"
+import { defaultStorePaymentCollectionFields } from "./query-config"
+import { StorePostPaymentCollectionsPaymentSessionReq } from "./validators"
+
+export const POST = async (
+  req: AuthenticatedMedusaRequest<StorePostPaymentCollectionsPaymentSessionReq>,
+  res: MedusaResponse
+) => {
+  const { id } = req.params
+  const { context, provider_id, data } = req.body
+
+  // If the customer is logged in, we auto-assign them to the payment collection
+  if (req.auth?.actor_id) {
+    context.customer = {
+      ...context.customer,
+      id: req.auth.actor_id,
+    }
+  }
+
+  const paymentService = req.scope.resolve<IPaymentModuleService>(
+    ModuleRegistrationName.PAYMENT
+  )
+
+  const paymentCollection = await paymentService.retrievePaymentCollection(id, {
+    select: ["id", "amount", "currency_code"],
+  })
+
+  const workflowInput = {
+    payment_collection_id: id,
+    provider_id: provider_id,
+    data,
+    context,
+    amount: paymentCollection.amount, // Question for the future: How do we handle split payments?
```

Comment:
One way to handle split payments is to not really care whether a payment is split or not. Let me explain:
1. You create a draft order / order awaiting payment
2. You do a payment of whatever amount you want, as many times as you want. (we check that you don't overpay)
3. The order is marked as paid only when the order amount = the amounts you've paid for this particular order.

Like this, you remove all complexity of managing split payments as part of the payment module, and any orchestration you need to do there.

There are different approaches and I think this will require a written document with proposals, but this is roughly how I've worked with split payments before